src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -4322,6 +4322,17 @@ operand immI_cmp_cond()
   interface(CONST_INTER);
 %}
 
+// BoolTest condition for unsigned compare
+operand immI_cmpU_cond()
+%{
+  predicate(n->get_int() > (int)(BoolTest::unsigned_compare));
```

Comment:
This function is in x86.ad. I suggest you move it to shared code, then use it here.
```
static inline bool is_unsigned_booltest_pred(int bt) {
  return  ((bt & BoolTest::unsigned_compare) == BoolTest::unsigned_compare);
}

```